crates/miden-lib/src/account/auth/mod.rs

```diff
@@ -119,6 +121,25 @@ impl From<RpoFalcon512ProcedureAcl> for AccountComponent {
     }
 }
 
+/// An [`AccountComponent`] that provides no authentication - it only increments the nonce.
+///
+/// This component exports the procedure `auth__none`, which simply increments the account nonce
+/// without performing any signature verification. This is useful for testing scenarios or accounts
+/// that don't require authentication.
+///
+/// This component supports all account types.
+pub struct AuthNone;
```

Comment:
Should we name this still `NoAuth`? I know the name of the procedure is `auth__none`, but mostly because `auth__no_auth` looks a bit weird (though, maybe it is not too bad).